src/adv.py

```diff
@@ -33,19 +44,180 @@
+                    for i in room.keys():
+                        if to_rm_desc == room[i]:
+                            player_info.location = i
+        print('Current room: ', player_info.location)
+        print(f'You are currently in {player_info.location}:\n\nRoom description:\n{room[player_info.location]}\n\n')
+    elif direction == 'east' and hasattr(room[player_info.location], 'e_to') and (room[player_info.location].e_to != room[player_info.location]):        
+        if hasattr(room[player_info.location], 'e_to'):
+            # cur_rm_desc = room[player_info.location]
+            to_rm_desc = room[player_info.location].e_to
+            for i in room.keys():                
+                if player_info.location == i and hasattr(room[player_info.location], 'e_to'):                    
+                    for i in room.keys():
+                        if to_rm_desc == room[i]:
+                            player_info.location = i
+        print('Current room: ', player_info.location)
+        print(f'You are currently in {player_info.location}:\n\nRoom description:\n{room[player_info.location]}\n\n')
+    elif direction == 'west' and hasattr(room[player_info.location], 'w_to') and (room[player_info.location].w_to != room[player_info.location]):        
+        if hasattr(room[player_info.location], 'w_to'):
+            # cur_rm_desc = room[player_info.location]
+            to_rm_desc = room[player_info.location].w_to
+            for i in room.keys():                
+                if player_info.location == i and hasattr(room[player_info.location], 'w_to'):                    
+                    for i in room.keys():
+                        if to_rm_desc == room[i]:
+                            player_info.location = i
+        print('Current room: ', player_info.location)
+        print(f'You are currently in {player_info.location}:\n\nRoom description:\n{room[player_info.location]}\n\n')            
+    else: print("You can't go that way")
+# look function that checks room for items
+
+def look():
+    if len(room[player_info.location].items) == 0:
+        print('Oof dusty.')
+    else:
+        print('The horrors you see are: ')
+        for i in room[player_info.location].items:
+            print(i) 
+
+def take():
+    if len(room[player_info.location].items) == 0:
+        print("Room's been cleaned out. By sombody... \n or something! Or was it me? <<hiccup>>")
+    else:    
+        player_info.items = player_info.items + room[player_info.location].items
+        room[player_info.location].items = []
+        print("You've added some weight to your rucksack. (type 'items' to see what you have)")
+
+def check_bag():
+    if len(player_info.items) == 0:
+        print('nothing in here yet\nkeep your eyes peeled')
+    else:
+        print('Your items:')
+        for i in player_info.items:
+            print(i)
+    
+def drop():
```

Comment:
I think there is a bug in the drop function still.  It only display rucksack after I picked up some items and the program crashes when I enter '2'

Reply:
bug fixed!